packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,451 @@
+import { EntityManager } from "typeorm"
+import { AbstractBatchJobStrategy, IFileService } from "../interfaces"
+import { BatchJob, Product } from "../models"
+import { ProductService, BatchJobService } from "../services"
+import { BatchJobStatus } from "../types/batch-job"
+import { defaultAdminProductRelations } from "../api/routes/admin/products"
+import { ProductRepository } from "../repositories/product"
+import { FindConfig } from "../types/common"
+
+type Context = {
+  listConfig: FindConfig<Product>
+  filterableFields: Record<string, unknown>
+  offset: number
+  count: number
+  progress: number
+}
+
+type InjectedDependencies = {
+  manager: EntityManager
+  batchJobService: BatchJobService
+  productService: ProductService
+  fileService: IFileService<any>
+  productRepository: typeof ProductRepository
+}
+
+type DataGetter = (line: Product) => string
+
+export default class ProductExportStrategy extends AbstractBatchJobStrategy<ProductExportStrategy> {
+  public static identifier = "product-export-strategy"
+  public static batchType = "product-export"
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly batchJobService_: BatchJobService
+  protected readonly productService_: ProductService
+  protected readonly fileService_: IFileService<any>
+  protected readonly productRepository_: typeof ProductRepository
+
+  protected readonly relations_ = [...defaultAdminProductRelations]
+  protected readonly columnDescriptors = new Map<string, DataGetter>([
+    ["Product Handle", (line: Product): string => line.handle],
+    ["Product Title", (line: Product): string => line.title],
+    ["Product Subtitle", (line: Product): string => line.subtitle],
+    ["Product Description", (line: Product): string => line.description],
+    ["Product Status", (line: Product): string => line.status],
+    ["Product Thumbnail", (line: Product): string => line.thumbnail],
+    [
+      "Product [Weight, Length, Width, Height]",
+      (line: Product): string => {
+        const output = [
+          line.weight?.toString(),
+          line.length?.toString(),
+          line.width?.toString(),
+          line.height?.toString(),
+        ]
+        return output.some((v) => v) ? output.toString() : ""
+      },
+    ],
+    [
+      "Product [HS_CODE, Origin Country, Mid Code, Material]",
+      (line: Product): string => {
+        const output = [
+          line?.hs_code,
+          line?.origin_country,
+          line?.mid_code,
+          line?.material,
+        ]
+        return output.some((v) => v) ? output.toString() : ""
+      },
+    ],
```

Comment:
todo: my bad for not being clear about this but `Product [Weight, Length ...]` was really just an abbreviation for having all the columns written out. 

So this should in fact be four separate columns for the dimensions and four separate columns for the customs stuff :)

Reply:
No problem at all 👍